src/node.cc

```diff
@@ -1830,6 +1840,14 @@ void Init(std::vector<std::string>* argv,
   // Make inherited handles noninheritable.
   uv_disable_stdio_inheritance();
 
+#if defined(NODE_REPORT)
+  // Cache the vanilla command line to be
+  // used in diagnostic reports.
+  for (size_t i = 0; i < argv->size(); ++i) {
+    per_process_opts->cmdline += argv->at(i) + " ";
```

Comment:
I am breaking my head on this.

I am able to repro the issue with the  example you quoted. the result is that the command line appears without single quote and loosing the association between the two args.

Where is the issue? while appending to a std::string ? or while padding with spaces? or retrieving it from the string? Without figuring that out I am now sure how storing it in an array is going to solve it! Any pointers?

Reply:
@gireeshpunathil The issue is that you’re joining strings with spaces, when they can contain spaces themselves. Using an array would remove the distinction between spaces-inside-arguments and spaces-as-separators, because the latter ones don’t exit?

Reply:
@addaleax - agree, but in terms of represenating it in the report, what difference it makes? eventually we need to dump things from the array and compose as a single string?

I could take a deep copy of the vector's backing array and store it as such, but when reporting, still I need to compose with spaces. for example this is how we repsent today:

```
"commandLine": "./node --experimental-report --report-uncaught-exception w.js "
```

or do you mean to say we could actually represent this as an array? like the way we do for few sections such as stack, shared libs, etc.? pls let me know.

Reply:
@gireeshpunathil Why can’t we provide the user with `"commandLine": ["./node", "--experimental-report", ...]`?

Reply:
done, command line is an array now.